apps/wallet-mobile/src/features/Swap/common/AmountCard/AmountCard.tsx

```diff
@@ -144,78 +145,88 @@ const useStrings = () => {
   }
 }
 
-const styles = StyleSheet.create({
-  container: {
-    borderRadius: 8,
-    borderWidth: 1,
-    borderColor: COLORS.TEXT_GRAY3,
-    paddingTop: 16,
-    paddingBottom: 16,
-    paddingLeft: 16,
-    paddingRight: 16,
-    padding: 10,
-    height: 86,
-  },
-  borderError: {
-    borderColor: COLORS.ALERT_TEXT_COLOR,
-    borderWidth: 2,
-  },
-  active: {
-    borderWidth: 2,
-    borderColor: '#242838',
-  },
-
-  label: {
-    position: 'absolute',
-    top: -7,
-    left: 10,
-    backgroundColor: COLORS.WHITE,
-    paddingHorizontal: 5,
-    fontSize: 12,
-    color: COLORS.ERROR_TEXT_COLOR_DARK,
-  },
-  labelError: {
-    color: COLORS.ALERT_TEXT_COLOR,
-  },
-  content: {
-    display: 'flex',
-    flexDirection: 'row',
-    justifyContent: 'space-between',
-    height: 64,
-  },
-  amountInput: {
-    paddingVertical: 0,
-    minWidth: 120,
-    maxWidth: 200,
-    height: 34,
-    fontSize: 16,
-    color: '#000000',
-  },
-  amountWrapper: {
-    flex: 1,
-  },
-  rightSection: {
-    flexDirection: 'column',
-    justifyContent: 'flex-start',
-  },
-  sectionContainer: {
-    flexDirection: 'row',
-    alignSelf: 'flex-end',
-    alignItems: 'center',
-  },
-  coinName: {
-    fontSize: 16,
-    fontWeight: '400',
-  },
-  balanceText: {
-    fontSize: 12,
-    color: COLORS.TEXT_INPUT,
-  },
-  errorText: {
-    color: COLORS.ALERT_TEXT_COLOR,
-    fontSize: 12,
-  },
-  grayText: {
-    color: '#6B7384',
-  },
-})
+const useStyles = () => {
+  const {theme} = useTheme()
+  const {color} = theme
+  const styles = StyleSheet.create({
```

Comment:
Hi Michael, don't think that there will be issues because is a hook and react will automatically handle the change on dependency change 🤔

Reply:
Its possible to use variables in that:

```
const styles = StyleSheet.create({
  container: {
    ...
  },
  foo: bar => ({
    backgroundColor: bar,  
})
})
```

<FancyComponet style={styles.foo('fancy-color')}  />

Reply:
I wish I shared such an optimism on hooks having good performance. They tend to run way too many times.
In this case, `useStyles` uses `useTheme` inside, which is a context hook. But `useStyles` is not limited in any way on how many times it runs. `StyleSheet.create` will be called on every render.

I don't think we're tracking performance, or even that this in particular will be noticeable. But definitely it'll be more costly than the previous way.
I assume moving `theme` from being static to being inside hooks is because we want them reactive, to change to dark mode on the fly or something?

`StyleSheet.create` could be done inside a memo to reduce rerenders, but that also has some extra setup and memory cost. Hard to tell what's better without profiling it.